client/src/components/BecomeInterpreterForm.js

```diff
@@ -0,0 +1,164 @@
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+import { Button, Form, Dropdown, Grid } from "semantic-ui-react";
+
+const languageOptions = [
+  { key: "Ar", text: "Arabic", value: "Arabic" },
+  { key: "Chi", text: "Chinese", value: "Chinese" },
+  { key: "Dan", text: "Danish", value: "Danish" },
+  { key: "Dut", text: "Dutch", value: "Dutch" },
+  { key: "En", text: "English", value: "English" },
+  { key: "Fre", text: "French", value: "French" },
+  { key: "Ger", text: "German", value: "German" },
+  { key: "Grek", text: "Greek", value: "Greek" },
+  { key: "Hun", text: "Hungarian", value: "Hungarian" },
+  { key: "Ita", text: "Italian", value: "Italian" },
+  { key: "Jap", text: "Japanese", value: "Japanese" },
+  { key: "Kor", text: "Korean", value: "Korean" },
+  { key: "Lith", text: "Lithuanian", value: "Lithuanian" },
+  { key: "Per", text: "Persian", value: "Persian" },
+  { key: "Pol", text: "Polish", value: "Polish" },
+  { key: "Por", text: "Portuguese", value: "Portuguese" },
+  { key: "Rus", text: "Russian", value: "Russian" },
+  { key: "Spa", text: "Spanish", value: "Spanish" },
+  { key: "Swe", text: "Swedish", value: "Swedish" },
+  { key: "Tur", text: "Turkish", value: "Turkish" },
+  { key: "Viet", text: "Vietnamese", value: "Vietnamese" }
+];
+
+class BecomeInterpreterForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: "",
+      username: "",
+      password: "",
+      confirmationpassword: "",
+      languages: "",
+      userCreated: null
+    };
+  }
+
+  handleChange = event => {
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  };
+  handleSubmit = e => {
+    e.preventDefault();
+    const { password, confirmationpassword } = this.state;
+    // perform all neccassary validations
+    if (password !== confirmationpassword) {
+      alert("Passwords don't match");
+    } else {
+      // make API call
+      const interpreterRequest = {
+        method: "POST",
+        body: JSON.stringify({
+          email: this.state.email,
+          username: this.state.username,
+          password,
+          confirmationpassword,
+          languages: this.state.languages
+        }),
+        headers: {
+          "Content-Type": "application/json"
+        }
+      };
+      fetch("http://localhost:4000/auth/register", interpreterRequest)
+        .then(res => res.json())
+        .then(data => {
+          this.setState({
+            email: "",
+            username: "",
+            password: "",
+            confirmationpassword: "",
+            languages: "",
+            userCreated: true
+          });
```

Comment:
Can you extract this.setState to a different function that you can name resetForm for example.